main.js

```diff
@@ -1,158 +1,175 @@
-  On platforms that support high pixel density displays (such as Apple Retina), we can append @2x after image's base
-  filename to mark it as a 2x scale high resolution image. */
+  // Create Tray icon
   const iconPath = path.join(__dirname, "icons", "frogIconTemplate.png");
-  const icon = nativeImage.createFromPath(iconPath);
-  icon.resize({ width: 16, height: 16 });
-  let tray = new Tray(icon);
-
-  tray.on('click', () => {
-    window.isVisible() ? window.hide() : window.show();
-    tray.closeContextMenu();
-  })
-  tray.on('right-click', () => {
-    tray.popUpContextMenu(contextMenu)
-  })
-
-  //Add cont menu for click on Tray bar icon
+  const icon = nativeImage.createFromPath(iconPath).resize({ width: 16, height: 16 });
+  
+  tray = new Tray(icon);
+
+  tray.on("click", () => {
+    const currentView = mainView || welcomeView;
+    if (currentView && !currentView.isDestroyed()) {
+      if (currentView.isVisible()) {
+        //hide if window open
+        currentView.hide(); 
+      } else {
+        //show if window hide
+        currentView.show();
+        currentView.focus();
+      }
+    }
+  });
+  
   const contextMenu = Menu.buildFromTemplate([
     {
       label: "Close Frog-app",
       type: "normal",
       click: () => {
-        app.exit(0);
+        app.isQuitting = true; // for close app in cotext menu
+        if (mainView) {
+          mainView.close();
+        }
+        app.quit();
       },
     },
   ]);
 
-  //add Workaround for Linux for context menu
-  if (process.platform === "linux") {
-    tray.setContextMenu(contextMenu);
-  }
-  
+  tray.setContextMenu(contextMenu);
   tray.setToolTip("Frog-app.");
-});
 
-// macOS apps generally continue running even without any windows open.
-// Activating the app when no windows are available should open a new one.
-app.on("activate", () => {
-  if (BrowserWindow.getAllWindows().length === 0) createWindow();
+  app.on('activate', () => {
+    if (BrowserWindow.getAllWindows().length === 0) renderPasswordWindow();
+  });
+
+  // Path for dir with txt file
+  const userFilesDir = path.join(__dirname, "User files");
+
+  // Path for file in dir and add const
+  const filePath = path.join(userFilesDir, "user_file.txt");
+
+  // add dir 'User files', if its not
+  (function createFileIfNotExists() {
+    if (!fs.existsSync(userFilesDir)) {
+      fs.mkdirSync(userFilesDir); 
+    }
+
+    //Abd add file 'user_file', if its not
+    if (!fs.existsSync(filePath)) {
+      fs.writeFileSync(filePath, "", "utf8");
+
+    // Check console
+      console.log(`File created at: ${filePath}`);
+
+      //Add message with a greeting if the user has opened the application for the first time
+      dialog.showMessageBox(welcomeView, {
+        title: "Welcome to Frogg-app",
+        message: "File has been created.",
+        type: "info"
+      });
+    } else {
+    // Check console
+      console.log(`File already exists at: ${filePath}`);
```

Comment:
I'm not sure that is a useful info, we will see this log every time after the file will be created